### PR TITLE
metadata.table模块完成

### PR DESCRIPTION
**Pull request by bluepongo**
单元测试以及http测试已通过

### CODE REVIEW

**Review by romberli — CHANGES_REQUESTED** [review]

File: internal/app/metadata/table_entity.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+)
+
+var _ metadata.Table = (*TableInfo)(nil)
+
+// TableInfo include information of logic table
+type TableInfo struct {
+	TableRepo metadata.TableRepo
+	DBName    string `middleware:"db_name" json:"db_name"`
+	TableName string `middleware:"table_name" json:"table_name"`
+	// TableStatistics []metadata.TableStatistic `middleware:"table_statistics" json:"table_statistics"`
```
[the remark this answers is not in the document]
Comment (SpicyChickenFLY) [reply internal/app/metadata/table_entity.go]: Done in [a8eef56](https://github.com/romberli/das/pull/122/commits/a8eef562a56b6e7109a0bdd21b762b049773da2e)

File: api/v1/metadata/table.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+}
+
+// @Tags	Tables
+// @Summary analyze table by host info and db name and table name
+// @Accept	application/json
+// @Param	host_ip			body string	true "host ip"
+// @Param	port_num		body int	true "port num"
+// @Param	db_name			body string	true "db name"
+// @Param	table_name		body string	true "table name"
+// @Param	account_name	body string true "account name or employee id"
```
[the remark this answers is not in the document]
Comment (SpicyChickenFLY) [reply api/v1/metadata/table.go]: Done in [688e981](https://github.com/romberli/das/pull/122/commits/688e98135bf3dc4117cee11dc62fda19be9f21cd)

File: api/v1/metadata/table.go [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+	masterServers, err := mysqlCluster.GetMasterServers()
+	if err != nil {
+		resp.ResponseNOK(c, msgmeta.ErrMetadataGetMasterServers, mysqlCluster.Identity(), err)
+		return
+	}
+
+	hostIP := masterServers[constant.ZeroInt].GetHostIP()
+	portNum := masterServers[constant.ZeroInt].GetPortNum()
+	dbAddr := fmt.Sprintf("%s:%d", hostIP, portNum)
+	dbUser := global.DASMySQLPool.Config.DBUser
```
[the remark this answers is not in the document]
Comment (SpicyChickenFLY) [reply api/v1/metadata/table.go]: Done in [e457eb9](https://github.com/romberli/das/pull/122/commits/e457eb9cae2f114d1776dcab72119bac02a9939b)

File: api/v1/metadata/table.go [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+	masterServers, err := mysqlCluster.GetMasterServers()
+	if err != nil {
+		resp.ResponseNOK(c, msgmeta.ErrMetadataGetMasterServers, mysqlCluster.Identity(), err)
+		return
+	}
+	hostIP := masterServers[constant.ZeroInt].GetHostIP()
+	portNum := masterServers[constant.ZeroInt].GetPortNum()
+	dbAddr := fmt.Sprintf("%s:%d", hostIP, portNum)
+	dbUser := global.DASMySQLPool.Config.DBUser
+	dbPass := global.DASMySQLPool.Config.DBPass
```
[the remark this answers is not in the document]
Comment (SpicyChickenFLY) [reply api/v1/metadata/table.go]: Done in [e457eb9](https://github.com/romberli/das/pull/122/commits/e457eb9cae2f114d1776dcab72119bac02a9939b)

File: api/v1/metadata/table.go [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+	mysqlCluster := ds.GetMySQLCluster()
+	masterServers, err := mysqlCluster.GetMasterServers()
+	if err != nil {
+		resp.ResponseNOK(c, msgmeta.ErrMetadataGetMasterServers, mysqlCluster.Identity(), err)
+		return
+	}
+	hostIP := masterServers[constant.ZeroInt].GetHostIP()
+	portNum := masterServers[constant.ZeroInt].GetPortNum()
+	dbAddr := fmt.Sprintf("%s:%d", hostIP, portNum)
+	dbUser := global.DASMySQLPool.Config.DBUser
```
[the remark this answers is not in the document]
Comment (SpicyChickenFLY) [reply api/v1/metadata/table.go]: Done in [e457eb9](https://github.com/romberli/das/pull/122/commits/e457eb9cae2f114d1776dcab72119bac02a9939b)

File: api/v1/metadata/table.go [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+		resp.ResponseNOK(c, message.ErrFieldNotExists, tableDBIDJSON)
+		return
+	}
+	dbID := int(dbIDInterface.(float64))
+	tableNameInterface, tableNameExists := dataMap[tableNameJSON]
+	if !tableNameExists {
+		resp.ResponseNOK(c, message.ErrFieldNotExists, tableNameJSON)
+		return
+	}
+	tableName := tableNameInterface.(string)
```
[the remark this answers is not in the document]
Comment (SpicyChickenFLY) [reply api/v1/metadata/table.go]: changed to ShouldBindJSON in [f1e59f4](https://github.com/romberli/das/pull/122/commits/f1e59f452334e0ffbed94c78015a8ff76a6c9152)

**Pull request merged**